openssl/src/x509/mod.rs

```diff
@@ -761,7 +761,7 @@ impl X509 {
                     ffi::PEM_read_bio_X509(bio.as_ptr(), ptr::null_mut(), None, ptr::null_mut());
                 if r.is_null() {
                     let err = ffi::ERR_peek_last_error();
-                    if ffi::ERR_GET_LIB(err) as X509LenTy == ffi::ERR_LIB_PEM
+                    if ffi::ERR_GET_LIB(err) as X509LenTy == ffi::ERR_LIB_PEM.try_into().unwrap()
```

Comment:
What's this change for?

Reply:
I had a build error and cargo suggested me to make that change, it had a mismatch between `i32` and `u32`. I don't know what caused that, maybe the boringssl version? I built boringssl from commit `5d58c559ace6a24ea6613e412b26bd4c50668ab3`.

Reply:
It's not needed for either the version we're using in CI, nor for boringssl HEAD as far as I can tell, so it shouldn't be part of this PR

Reply:
I removed that and now got the same error
```
error[E0308]: mismatched types
   --> openssl\src\x509\mod.rs:764:62
    |
764 |                     if ffi::ERR_GET_LIB(err) as X509LenTy == ffi::ERR_LIB_PEM
    |                        ----------------------------------    ^^^^^^^^^^^^^^^^ expected `u32`, found `i32`
    |                        |
    |                        expected because this is `u32`
    |
help: you can convert an `i32` to a `u32` and panic if the converted value doesn't fit
    |
764 |                     if ffi::ERR_GET_LIB(err) as X509LenTy == ffi::ERR_LIB_PEM.try_into().unwrap()
    |                                                                              ++++++++++++++++++++
```
That BoringSSL version is used by Python's cryptography 0.42